### PR TITLE
Sumarize project, ready to PR

### PR DESCRIPTION
**Pull request by giladyosef54**

### CODE REVIEW

File: app/summurizing_task/BST.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        if self._root is None:
+            self._root = BinaryTree.Node(value)
+        else:
+            self._insert(value, self._root)
+
+    def validate(self):
+        return self._validate_subtree()
+
+    @staticmethod
+    def _insert(value, node, node_type = BinaryTree.Node):
```
Comment (MayaLandman) [inline app/summurizing_task/BST.py]: not in use

File: app/summurizing_task/BinaryTree.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+from abc import abstractmethod, ABC
+import pymongo
```
Comment (MayaLandman) [inline app/summurizing_task/BinaryTree.py]: uneccerry import (not in use)

File: app/summurizing_task/BinaryTree.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+
+class BinaryTree(ABC):
+    class Node:
+        def __init__(self, value):
+            self.left = None
+            self.right = None
+            self.value = value
```
Comment (MayaLandman) [inline app/summurizing_task/BinaryTree.py:12]: consistent name use either value or _value

File: app/summurizing_task/BinaryTree.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+        @property
+        def value(self):
```
Comment (MayaLandman) [inline app/summurizing_task/BinaryTree.py]: unnecessary, access the value directly instead

File: app/summurizing_task/BinaryTree.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            self._value = value
+
+    def __init__(self):
+        self._root = None
+
+    def __iter__(self):
+        return InOrderIterator(self._root)
+
+    @abstractmethod
+    def insert(self, value):
```
Comment (MayaLandman) [inline app/summurizing_task/BinaryTree.py:21]: not in use Remove it if not needed

File: app/summurizing_task/BinaryTree.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        return self._get_in_order_values(self._root)
+
+    def get_pre_order_values(self):
+        return self._get_pre_order_values(self._root)
+
+    def get_post_order_values(self):
+        return self._get_post_order_values(self._root)
+
+    @staticmethod
+    def _get_in_order_values(node, values_list = list()):
```
Comment (MayaLandman) [inline app/summurizing_task/BinaryTree.py]: use default value instead of mutable default value

File: app/summurizing_task/BinaryTree.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        if node:
+            BinaryTree._get_in_order_values(node.left, values_list)
+            values_list.append(node.value)
+            BinaryTree._get_in_order_values(node.right, values_list)
+            return values_list
+
+    @staticmethod
+    def _get_pre_order_values(node, values_list=list()):
```
Comment (MayaLandman) [inline app/summurizing_task/BinaryTree.py]: same

File: app/summurizing_task/BinaryTree.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        if node:
+            values_list.append(node.value)
+            BinaryTree._get_pre_order_values(node.left, values_list)
+            BinaryTree._get_pre_order_values(node.right, values_list)
+            return values_list
+
+    @staticmethod
+    def _get_post_order_values(node, values_list=list()):
```
Comment (MayaLandman) [inline app/summurizing_task/BinaryTree.py]: same

File: app/summurizing_task/ds_to_db.py [hunk 9]
```diff
@@ -0,0 +1,112 @@
+import pymongo
+import AVLTree
+
+
+collections_metadata_col = 'dwarfs'
+treasures_and_keepers_db = 'Treasures_and_keepers'
+host_url = "mongodb://localhost:27017/"
```
Comment (MayaLandman) [inline app/summurizing_task/ds_to_db.py]: Global constants should be in uppercase

File: app/summurizing_task/ds_to_db.py [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        field = 'name'
+        query = {field: self._dwarf_name}
+        if dwarfs.count_documents(query) == 0:
+            new_dwarf = {'_id': dwarfs_num, field: self._dwarf_name}
+            dwarfs.insert_one(new_dwarf)
+
+        self._dwarf_col = self._db[self._dwarf_name]
+        self._treasure_field = 'value'
+        for treasure in self._dwarf_col.find({}, {self._treasure_field: 1}):
+            super().insert(treasure[self._treasure_field])
```
Comment (MayaLandman) [inline app/summurizing_task/ds_to_db.py]: can be super().insert(treasure['value'])

File: app/summurizing_task/AVLTree.py [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+
+class AVLTree(BST):
+    class AVLNode(BinaryTree.Node):
+
+        def __init__(self, value):
+            super().__init__(value)
+            self.parent = None
+            self.height = 1
+
+        def update_height(self):
```
Comment (NadavGlazer) [inline app/summurizing_task/AVLTree.py:14]: Add documantation to every function
Read about the python syntax

File: app/summurizing_task/AVLTree.py [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+    def insert(self, value):
+        if self._root is None:
+            self._root = AVLTree.AVLNode(value)
+        else:
+            self._root = self._insert_to_node(value, self._root)
+
+    def delete(self, value):
+        self._root = AVLTree._delete(self._root, value)
+
+    def exist(self, value):
```
Comment (NadavGlazer) [inline app/summurizing_task/AVLTree.py:47]: Better name

File: app/summurizing_task/ds_to_db.py [hunk 13]
```diff
@@ -0,0 +1,112 @@
+import pymongo
+import AVLTree
+
+
+COLLECTIONS_METADATA_COL = 'dwarfs'
```
Comment (NadavGlazer) [inline app/summurizing_task/ds_to_db.py]: Move to enviroment variable

File: app/summurizing_task/MongoBackedAVL.py [hunk 14]
```diff
@@ -0,0 +1,77 @@
+from dotenv import load_dotenv
+import AVLTree
+import pymongo
+from mongoCollectionManager import MongoCollectionManager
+from os import getenv
+
+load_dotenv()
+
+
+TREASURES_AND_KEEPERS_DB = getenv("DATA_BASE")
```
Comment (NadavGlazer) [inline app/summurizing_task/MongoBackedAVL.py]: Create a file name config.py where you extract all the env variables

**Pull request closed without merging**